src/app/parcels/ActionBar/Actions.tsx

```diff
@@ -179,22 +179,32 @@ const ActionsButton: React.FC<ActionsButtonProps> = ({
 };
 
 interface Props {
-    selectedParcels: ParcelsTableRow[];
+    fetchSelectedParcels: (checkedParcelIds: string[]) => Promise<ParcelsTableRow[]>;
     onDeleteParcels: (parcels: ParcelsTableRow[]) => void;
     actionAnchorElement: HTMLElement | null;
     setActionAnchorElement: React.Dispatch<React.SetStateAction<HTMLElement | null>>;
     modalError: string | null;
     setModalError: React.Dispatch<React.SetStateAction<string | null>>;
+    parcelIds: string[];
 }
 
 const Actions: React.FC<Props> = ({
-    selectedParcels,
+    fetchSelectedParcels,
     onDeleteParcels,
     actionAnchorElement,
     setActionAnchorElement,
     modalError,
     setModalError,
+    parcelIds,
 }) => {
+    const [selectedParcels, setSelectedParcels] = useState<ParcelsTableRow[]>([]);
+
+    useEffect(() => {
+        (async () => {
+            setSelectedParcels(await fetchSelectedParcels(parcelIds));
```

Comment:
Okay I have added an error message that is shown to the user in this case. Like how "please select at least 1 parcel" is shown to the user. I've changed the code so that the fetchSelectedParcels call is made only when a status/action is selected from their dropdown. This is what I was aiming to do, but it wasn't working - the call was being made whenever parcels were selected/unselected -  now its working as I wanted.